app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageFragment.kt

```diff
@@ -432,16 +429,39 @@ class NewMessageFragment : Fragment() {
         }
     }
 
-    private fun observeNewAttachments() = with(newMessageViewModel) {
-        importedAttachments.observe(viewLifecycleOwner) { (attachments, importationResult) ->
+    private fun observeAttachments() {
+        newMessageViewModel.attachmentsLiveData.observe(viewLifecycleOwner) { attachments ->
+
+            if (shouldRegisterToImportedAttachments) observeImportAttachments()
+
+            attachmentAdapter.submitList(attachments)
+
+            binding.attachmentsRecyclerView.isVisible = if (attachments.isEmpty()) {
+                if (attachmentAdapter.itemCount != 0) TransitionManager.beginDelayedTransition(binding.root)
```

Comment:
I don't really get the use of this check, as you submit the adapter with "attachments" before. So if 'attachments' is empty, this will be == 0 no ?

Reply:
This code is wrong, here is the correct version :
```kt
private fun observeAttachments() = with(newMessageViewModel) {
    attachmentsLiveData.observe(viewLifecycleOwner) { attachments ->

        if (shouldRegisterToImportedAttachments) observeImportAttachments()

        val shouldTransition = attachmentAdapter.itemCount != 0 && attachments.isEmpty()
        attachmentAdapter.submitList(attachments)
        if (shouldTransition) TransitionManager.beginDelayedTransition(binding.root)
        binding.attachmentsRecyclerView.isVisible = attachments.isNotEmpty()

        updateIsSendingAllowed(attachments)
    }
}
```